engine/access/state_stream/backend/backend_executiondata_test.go

```diff
@@ -335,12 +352,50 @@ func (s *BackendExecutionDataSuite) TestGetExecutionDataByBlockID() {
 	// notify backend block is available
 	s.highestBlockHeader = block.ToHeader()
 
+	executionNodes := unittest.IdentityListFixture(2, unittest.WithRole(flow.RoleExecution))
+
 	var err error
 	s.Run("happy path TestGetExecutionDataByBlockID success", func() {
 		result.ExecutionDataID, err = s.eds.Add(ctx, execData.BlockExecutionData)
 		require.NoError(s.T(), err)
 
-		res, err := s.backend.GetExecutionDataByBlockID(ctx, block.ID())
+		metadata := &accessmodel.ExecutorMetadata{
+			ExecutionResultID: result.ID(),
+			ExecutorIDs:       executionNodes.NodeIDs(),
+		}
+
+		s.executionResultProvider.
+			On("ExecutionResultInfo", mock.Anything, mock.Anything).
+			Return(&optimistic_sync.ExecutionResultInfo{
+				ExecutionResultID: result.ID(),
+				ExecutionNodes:    executionNodes.ToSkeleton(),
+			}, nil).
+			Once()
+
+		s.executionStateCache.
+			On("Snapshot", mock.Anything).
```

Comment:
assert values where practical
```suggestion
			On("Snapshot", result.ID()).
```